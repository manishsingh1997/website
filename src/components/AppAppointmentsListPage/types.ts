import {HouseType} from '../types';

// TODO: Improve typing. And some of these types may be defined elsewhere
type AppointmentOrderProduct = {
  id: number,
  name: string,
  short_name: string,
  slug: string,
  is_active: boolean,
}

export type AppointmentOrder = {
  id: number,
  ordered_at: string, // It's actually a timestamp
  house: HouseType,
  product: AppointmentOrderProduct,
}

export type Appointment = {
  agent_name: string,
  date: string, // Perhaps we should use some date type
  id: number,
  order: AppointmentOrder,
  time_end: string, // Perhaps we should use some time type
  time_start: string, // Perhaps we should use some time type
  type: string,
}

export enum AppointmentsFilter {
  UPCOMING = 'Upcoming',
  PAST = 'Past',
  ALL = 'All',
}

// Already exists in erg-core-components
export type OptionType = {
  icon?: string;
  label: string;
  value: string;
  onClick?: () => void;
}
