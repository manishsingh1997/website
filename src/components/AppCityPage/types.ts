export type City = {
  slug: string,
  city: string,
  county: string,
  state:
    | 'AL' | 'AK' | 'AZ' | 'AR' | 'CA' | 'CO' | 'CT' | 'DE' | 'DC' | 'FL' | 'GA' | 'HI' | 'ID'
    | 'IL' | 'IN' | 'IA' | 'KS' | 'KY' | 'LA' | 'ME' | 'MD' | 'MA' | 'MI' | 'MN' | 'MS' | 'MO'
    | 'MT' | 'NE' | 'NV' | 'NH' | 'NJ' | 'NM' | 'NY' | 'NC' | 'ND' | 'OH' | 'OK' | 'OR' | 'PA'
    | 'RI' | 'SC' | 'SD' | 'TN' | 'TX' | 'UT' | 'VT' | 'VA' | 'WA' | 'WV' | 'WI' | 'WY',
  address?: string,
  phone: string,
  header: {
    title: string,
    bullets: string[],
  },
  blog: {
    articles: Array<{
      url: string,
      img: string,
      description: string,
    }>,
  },
  license: {
    url?: string,
  },
  licenses?: string[],
  packages: {
    title: string,
    description: string,
    data: Record<'wood' | 'vinyl' | 'chain-link' | 'box-wire', Array<{
      title: string,
      description: string,
      img: string,
    }>>,
  },
  projects: Array<{
    url?: string,
    img: string,
    label: string,
  }>,
  product: {
    type: 'Fence',
  },
  rating: Record<'Angi' | 'BBB' | 'Google' | 'Thumbtack' | 'Yelp', string>,
  review: Record<'Yelp' | 'Google', {
    title: string,
    name?: string,
    description?: string,
    address?: string,
    img?: string,
    url?: string,
  }>,
  regulations: {
    title: string,
    description: string,
    pdf?: string,
    img?: string,
    faq: Record<'front-yard' | 'side-yard' | 'rear-max-height' | 'setback' | 'corner-lot-rules', string>,
  },
};
