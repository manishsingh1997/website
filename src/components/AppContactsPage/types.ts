export type PrimaryContact = {
  id: number;
  type?: string;
  identifier: string;
  formatted_identifier: string;
  label: string | null;
  action?: 'add' | 'edit';
  error?: string;
};

export type Contact = {
  id: number;
  full_name: string;
  relation: string;
  primary_email: PrimaryContact;
  primary_phone: PrimaryContact;
  additional_emails: PrimaryContact[];
  additional_phones: PrimaryContact[];
  is_primary: boolean;
};

export type InitialContact = {
    id?: number,
    name?: string,
    email?: string,
    phone?: string,
}

export type AdditionalContact = PrimaryContact[];
