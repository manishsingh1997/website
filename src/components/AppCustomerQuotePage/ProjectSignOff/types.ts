
export type SignType = 'draw' | 'text';

export type SignData = {
  value: string; 
  type: SignType, 
  signedDate: string
  signedPDF?: string
};

export type Error = {
  text?: boolean;
  draw?: boolean;
  message?: string;
};

export type Order = {
  orderId: string;
  customerName: string;
  customerAddress: string;
  quoteDate: string;
}

export type SignoffStampType = {
  signoff_at: null | string,
  signoff_by: null | string | number,
  signoff_pdf: null | string,
  signoff_img: string,
}

export type ProjectSignOffProps = {
  isSigned: boolean;
  signedDate?: string;
  pdfURL?: string;
}
