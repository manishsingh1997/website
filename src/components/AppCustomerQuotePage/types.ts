import {ReactElement} from 'react';

// eslint-disable-next-line import/named
import {match} from 'react-router-dom';

import {AuthState} from '../../flux/reducers/auth';
import {Quote, QuoteLine} from '../types';

type Layout = {
  phoneNumber: string;
};

export type Params = {
  secret: string;
  path: string;
  quoteId?: string;
};

type BasePageProps = {
  match: match<Params>;
  auth: AuthState;
};

export type SignatureType = 'draw' | 'text';

export type SignatureData = {
  value: string;
  type: SignatureType;
  signedDate: string;
  signedPDF?: string;
};

export type OrderData = {
  orderId: string;
  customerName: string;
  customerAddress: string;
  quoteDate: string;
};

type Address = {
  formatted_address: string;
  zip_code: string;
  latitude: number;
  longitude: number;
};

export type Customer = {
  id: number;
  user_id: number;
  gid: string;
  full_name: string;
  email: string;
  phone_number: string;
  main_address: Address;
};

export type PaymentMethod = {
  card_last_digits: string;
  card_expiration_date: string;
  card_expiration_date_short: string;
};

export interface QuoteApproval {
  id: number;
  customer: Customer;
  is_primary: boolean;
  total_price: string;
  reviewed_at: string;
  approved_at: unknown;
  secret: string;
  quote: Quote;
  quote_approval_lines: QuoteLine[];
  payment_method: unknown;
  other_quote_approvals: unknown[];
  new_quote_approval: Quote | null;
  last_approved_project_revision_quote_approval: Quote | null;
  customer_pdf: string;
  contract_pdf: string;
  merged_customer_pdf: string;
  previously_approved_price: string;
  project_total_price: string;
  signoff_at: string;
  signoff_by: string;
  signoff_pdf: string;
  signoff_img: string;
}

export type AppCustomerQuotePageProps = {
  setPDFHeaderPhoneAction: (value: number) => void;
  layout: Layout;
  quoteFallback?: ReactElement;
} & BasePageProps;

export type QuoteBodyProps = {
  approveQuoteApproval: (stripeToken: string) => void;
  customerGID: string;
  paymentMethod: PaymentMethod | null;
  paymentMethodError: string | null;
  quoteApproval: QuoteApproval;
  shouldShowSignoffComponents?: boolean;
} & BasePageProps;

type SignOffCommonProps = {
  customerSignOffData: OrderData;
  isCustomerSigned: boolean;
  isSignLoading: boolean;
  onSubmitSignature: (value: string, type: string) => void;
  signatureData: SignatureData;
};

export type DefaultContentProps = QuoteBodyProps & SignOffCommonProps;

export type SignOffPdfProps = {
  asPDF: boolean;
  quoteApproval: QuoteApproval;
} & SignOffCommonProps;
