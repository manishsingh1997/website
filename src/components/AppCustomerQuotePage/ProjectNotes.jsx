import React from 'react';
import PropTypes from 'prop-types';

export default class ProjectNotes extends React.Component {
  static propTypes = {
    quote: PropTypes.object,
  };

  render() {
    const {quote} = this.props;
    return (
      <React.Fragment>
        {quote['description_html'] && (
          <div className="quote-notes card padding-40 spacing after__is-30 page-break">
            <h4>Project notes</h4>
            <div className="quote-projects-notes">{
              // eslint-disable-next-line react/no-danger
            } <div dangerouslySetInnerHTML={{__html: quote['description_html']}}/>
            </div>
          </div>
        )}
      </React.Fragment>);
  }
}
