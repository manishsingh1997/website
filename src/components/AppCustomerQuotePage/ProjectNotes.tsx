import React from 'react';

import {ProjectDetailsProps} from './types';

const ProjectNotes = ({quote}: ProjectDetailsProps) => (
  <div className="quote-notes card padding-40 spacing after__is-30 page-break">
    <div className="heading-md-text">Project notes</div>
    <div className="quote-projects-notes">
      <div dangerouslySetInnerHTML={{__html: quote.description_html}} />
    </div>
  </div>
);

export default ProjectNotes;
