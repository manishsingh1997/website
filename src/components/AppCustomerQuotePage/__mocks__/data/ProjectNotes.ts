/* eslint-disable max-len */
export default {
  quote: {
    description_html:
      '<ul><li><div><b>Why choose us?</b><br></div><div><b><br></b></div><ul><li>We guarantee the quality of our work through our industry leading warranty and source high quality materials to ensure a long lasting construction.&nbsp;</li><li>You will have an assigned project manager to coordinate your installation and our customer service desk will be available to assist Monday through Sunday.</li></ul><div><br></div><div><b>Important things to Consider:<br><br></b></div><ul><li>All lumber sizes are stated in nominal dimensions not actual measurements, this is part of the industry standard, please allow a small tolerance in case there is a few inches difference in the final measurements or fence height.</li></ul><div><br></div><ul><li>As part of our due diligence we request a dig clearance to identify underground public utility pipes, avoiding any unwanted incidents.<br>Please make sure to verify and point out property boundaries, buried private lines (gas, water or electric) or objects and sprinklers.</li></ul><div><br></div><ul><li>Lastly, please, arrange a 2ft clearance is provided along the fence line (cutting back vegetation and removing objects such as: household articles, river rocks, mulch among others) otherwise the project may be delayed causing additional charges.</li></ul></li></ul><div><br>',
  },
};
