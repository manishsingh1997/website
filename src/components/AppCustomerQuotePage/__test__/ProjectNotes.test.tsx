import React from 'react';

import {render, screen} from '@testing-library/react';

import '@testing-library/jest-dom';
import projectsNotesQuote from '../__mocks__/data/ProjectNotes';
import ProjectNotes from '../ProjectNotes';
import {ProjectDetailsProps} from '../types';

const mockProjectNotesQuoteProps = projectsNotesQuote as ProjectDetailsProps;

describe('Project Notes', () => {
  it('Should match heading and snapshot', () => {
    const {container} = render(<ProjectNotes quote={mockProjectNotesQuoteProps.quote} />);
    expect(screen.getByText('Project notes').matches('Project notes'));
    expect(container).toMatchSnapshot();
  });
});
