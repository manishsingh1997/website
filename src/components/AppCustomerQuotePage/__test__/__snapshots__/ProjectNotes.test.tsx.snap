// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Notes Should match heading and snapshot 1`] = `
<div>
  <div
    class="quote-notes card padding-40 spacing after__is-30 page-break"
  >
    <div
      class="heading-md-text"
    >
      Project notes
    </div>
    <div
      class="quote-projects-notes"
    >
      <div>
        <ul>
          <li>
            <div>
              <b>
                Why choose us?
              </b>
              <br />
            </div>
            <div>
              <b>
                <br />
              </b>
            </div>
            <ul>
              <li>
                We guarantee the quality of our work through our industry leading warranty and source high quality materials to ensure a long lasting construction.Â 
              </li>
              <li>
                You will have an assigned project manager to coordinate your installation and our customer service desk will be available to assist Monday through Sunday.
              </li>
            </ul>
            <div>
              <br />
            </div>
            <div>
              <b>
                Important things to Consider:
                <br />
                <br />
              </b>
            </div>
            <ul>
              <li>
                All lumber sizes are stated in nominal dimensions not actual measurements, this is part of the industry standard, please allow a small tolerance in case there is a few inches difference in the final measurements or fence height.
              </li>
            </ul>
            <div>
              <br />
            </div>
            <ul>
              <li>
                As part of our due diligence we request a dig clearance to identify underground public utility pipes, avoiding any unwanted incidents.
                <br />
                Please make sure to verify and point out property boundaries, buried private lines (gas, water or electric) or objects and sprinklers.
              </li>
            </ul>
            <div>
              <br />
            </div>
            <ul>
              <li>
                Lastly, please, arrange a 2ft clearance is provided along the fence line (cutting back vegetation and removing objects such as: household articles, river rocks, mulch among others) otherwise the project may be delayed causing additional charges.
              </li>
            </ul>
          </li>
        </ul>
        <div>
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
